diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ac95a587..941873719 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,7 +139,7 @@ SET_TARGET_PROPERTIES (sac2c sac4c sac2tex
 # This target will cause all build dirs (including externalprojects) to be cleaned
 ADD_CUSTOM_TARGET(fullclean
   COMMAND
-    +${CMAKE_COMMAND} --build ${PROJECT_BINARY_DIR} --target clean
+    ${CMAKE_COMMAND} --build ${PROJECT_BINARY_DIR} --target clean
   COMMENT "Cleaning sac2c and runtime libraries")
 
 # we need to convert the target `list' into a string in order to pass it to the
@@ -182,7 +182,7 @@ UNSET (BINARY_DIR) # make sure we don't cause a conflict
 # Create a step to clean the build directory
 ExternalProject_Add_Step (runtime_libraries clean
   COMMAND
-  +${CMAKE_COMMAND} --build ${RUNTIME_BINARY_DIR} --target clean
+  ${CMAKE_COMMAND} --build ${RUNTIME_BINARY_DIR} --target clean
   DEPENDEES configure
   COMMENT "Cleaning runtime libraries, in ${RUNTIME_BINARY_DIR}"
   EXCLUDE_FROM_MAIN 1
@@ -207,11 +207,16 @@ ExternalProject_Add_StepTargets (runtime_libraries configure build install)
 ExternalProject_Add_StepDependencies (runtime_libraries install runtime_libraries-build)
 ExternalProject_Add_StepDependencies (runtime_libraries build runtime_libraries-configure)
 FOREACH (__target ${RT_TARGETS})
+  STRING (REPLACE "_" "-"  __sbi "${__target}") # FIXME not consistent
+
+  MESSAGE(STATUS "Byproduct is: ${RUNTIME_BINARY_DIR}/lib/host/${__sbi}/libsac${BUILD_TYPE_POSTFIX}${SHARED_LIB_EXT}"
+  )
+
   ADD_CUSTOM_TARGET (libsac-${__target}
-    COMMAND +${CMAKE_COMMAND} --build ${RUNTIME_BINARY_DIR} --target libsac-${__target}
+    COMMAND ${CMAKE_COMMAND} --build ${RUNTIME_BINARY_DIR} --target libsac-${__target}
+    BYPRODUCTS "${RUNTIME_BINARY_DIR}/lib/host/${__sbi}/libsac${BUILD_TYPE_POSTFIX}${SHARED_LIB_EXT}"
   )
   ADD_DEPENDENCIES (libsac-${__target} runtime_libraries-configure)
-  STRING (REPLACE "_" "-"  __sbi "${__target}") # FIXME not consistent
   SET_TARGET_PROPERTIES (libsac-${__target}
     PROPERTIES
     OUTPUT_NAME "${RUNTIME_BINARY_DIR}/lib/host/${__sbi}/libsac${BUILD_TYPE_POSTFIX}${SHARED_LIB_EXT}"
