diff --git a/cmake/sac2c-version-related.cmake b/cmake/sac2c-version-related.cmake
index b3931c911..2714668b4 100644
--- a/cmake/sac2c-version-related.cmake
+++ b/cmake/sac2c-version-related.cmake
@@ -38,7 +38,7 @@ IF (NOT SAC2C_IS_DIRTY)
         WORKING_DIRECTORY
             ${PROJECT_SOURCE_DIR}
         RESULT_VARIABLE
-            SAC2C_IS_DIRTY)
+            _SAC2C_IS_DIRTY)
 ENDIF ()
 
 # This set a hook on the configuration system
diff --git a/cmake/sac2c/config.cmake b/cmake/sac2c/config.cmake
index d7c6734a0..066ce4e11 100644
--- a/cmake/sac2c/config.cmake
+++ b/cmake/sac2c/config.cmake
@@ -27,12 +27,14 @@ INCLUDE (CheckCCompilerFlag)
 # mark the state of the repository as dirty, as we don't allow
 # non-standard compilations of sac2c.
 IF (CMAKE_C_FLAGS)
-  SET (SAC2C_IS_DIRTY 1)
+  SET (_SAC2C_IS_DIRTY 1)
   MESSAGE (STATUS
            "NOTE: Additional C flags passed; changing the state of the "
            "build to dirty; no install targets will be generated.")
 ENDIF ()
 
+SET (SAC2C_IS_DIRTY 0)
+
 # Include other config files
 INCLUDE ("cmake/sac2c-version-related.cmake")
 
